ts/src/test/Exchange/base/test.sharedMethods.ts

```diff
@@ -396,6 +400,15 @@ function assertRoundMinuteTimestamp (exchange: Exchange, skippedProperties: obje
     assert (Precise.stringMod (ts, '60000') === '0', 'timestamp should be a multiple of 60 seconds (1 minute)' + logText);
 }
 
+function deepEqual (a: any, b: any) {
+    return JSON.stringify (a) === JSON.stringify (b);
```

Comment:
only c# needed some changes to support `JSON.`, e.g. adding custom `json` method in `base.helpers.cs`